be/btree.c

```diff
@@ -59,6 +59,104 @@ M0_INTERNAL const struct m0_fid_type m0_btree_fid_type = {
 	.ft_name = "btree fid",
 };
 
+struct btree_pair_ver {
+	uint64_t bpv_ver;
+};
+
+/*
+ * The key pointer is aligned with sizeof(void*). We add the version
+ * right before the key. This assertion helps us to ensure that this
+ * addition does not break this requirement for key alignment.
+ * See ::btree_save for details.
+ */
+M0_BASSERT(M0_IS_8ALIGNED(sizeof(struct btree_pair_ver)));
+
+enum {
+	/* Tombstone flag: marks a dead kv pair. We use the MSB here. */
+	BPV_TBS = 1L << (sizeof(uint64_t) * CHAR_BIT - 1),
+
+	/*
+	 * n_inf/p_inf marks non-allowed values of the
+	 * btree_pair_ver::bpv_ver.
+	 */
+	BPV_VER_PINF = UINT64_MAX & ~BPV_TBS,
+	BPV_VER_NINF = 0,
+
+	/* The biggest possible value of a version */
+	BPV_VER_MAX = BPV_VER_PINF - 1,
+	/* The lowest possible value of a version */
+	BPV_VER_MIN = BPV_VER_NINF + 1,
+};
+
+#define VER_DATUM_SIZE (sizeof(struct btree_pair_ver))
+M0_BASSERT(VER_DATUM_SIZE == sizeof(void *));
```

Comment:
this needs a comment

Reply:
I have removed this line; only the alignment requirement is still there.